contracts/pair_anchor/tests/integration.rs

```diff
@@ -0,0 +1,414 @@
+            astroport_pair_anchor::mock_anchor_contract::instantiate,
+            astroport_pair_anchor::mock_anchor_contract::query,
+        )
+        .with_reply_empty(astroport_pair_anchor::mock_anchor_contract::reply),
+    );
+
+    app.store_code(factory_contract)
+}
+
+#[test]
+fn test_compatibility_of_pair_anchor_with_routeswap() {
+    let bank = BankKeeper::new();
+    let mut app = mock_app(bank);
+
+    let owner = Addr::unchecked(OWNER);
+    let alice_address = Addr::unchecked("alice");
+
+    app.init_bank_balance(
+        &alice_address,
+        vec![
+            Coin {
+                denom: "uusd".to_string(),
+                amount: Uint128::new(2_000_000_000u128),
+            },
+            Coin {
+                denom: "uluna".to_string(),
+                amount: Uint128::new(10_000_000_000u128),
+            },
+        ],
+    )
+    .unwrap();
+
+    let token_code_id = store_token_code(&mut app);
+    let factory_code_id = store_factory_code(&mut app);
+    let router_code_id = store_router_code(&mut app);
+    let pair_anchor_code_id = store_pair_anchor_code(&mut app);
+    let pair_luna_code_id = store_pair_code(&mut app);
+    let anchor_code_id = store_anchor_code(&mut app);
+
+    let init_anchor = AnchorInstantiateMsg {};
+
+    let anchor_contract = app
+        .instantiate_contract(
+            anchor_code_id,
+            owner.clone(),
+            &init_anchor,
+            &[],
+            "ANCHOR",
+            None,
+        )
+        .unwrap();
+
+    let token_name = "aUST";
+
+    let init_token_msg = TokenInstantiateMsg {
+        name: token_name.to_string(),
+        symbol: token_name.to_string(),
+        decimals: 6,
+        initial_balances: vec![Cw20Coin {
+            address: alice_address.to_string(),
+            amount: Uint128::from(10_000_000_000u128),
+        }],
+        mint: Some(MinterResponse {
+            minter: anchor_contract.to_string(),
+            cap: None,
+        }),
+    };
+
+    let token_aust_contract = app
+        .instantiate_contract(
+            token_code_id,
+            owner.clone(),
+            &init_token_msg,
+            &[],
+            token_name,
+            None,
+        )
+        .unwrap();
+
+    let msg = AnchorExecuteMsg::SetToken(token_aust_contract.to_string());
+
+    app.execute_contract(owner.clone(), anchor_contract.clone(), &msg, &[])
+        .unwrap();
+
+    let init_factory = FactoryInstantiateMsg {
+        fee_address: None,
+        pair_configs: vec![
+            PairConfig {
+                code_id: pair_luna_code_id,
+                maker_fee_bps: 0,
+                pair_type: PairType::Xyk {},
+                total_fee_bps: 0,
+                is_disabled: false,
+                is_generator_disabled: false,
+            },
+            PairConfig {
+                code_id: pair_anchor_code_id,
+                maker_fee_bps: 0,
+                pair_type: PairType::Custom("anchor".to_string()),
```

Comment:
here I registered it as a custom pair type: anchor.
Also I dont know what should be set for is_generator_disabled.

Reply:
done